projects/max14906/src/examples/basic/basic_example.c

```diff
@@ -71,47 +71,47 @@ int basic_example_main()
 	enum max14906_climit climit = MAX14906_CL_300;
 	enum max14906_climit climit2 = MAX14906_CL_600;
 
-	/** MAX14906 Initialization */
-	ret = max14906_init(&max14906_desc, &max14906_ip);
-	if (ret)
-		goto exit;
+	struct no_os_callback_desc gpio_cb = {
+		.callback = gpio_callback_fn,
+		/** Parameter to be passed when the callback is called. */
+		.ctx = max14906_desc,
+		.event = NO_OS_EVT_GPIO,
+		.peripheral = NO_OS_GPIO_IRQ,
+		.handle = NULL
+	};
 
 	/** Global Interrupt Controller */
-	struct no_os_irq_ctrl_desc *global_desc;
-	struct no_os_irq_init_param global_desc_param = {
+	struct no_os_irq_ctrl_desc *gpio_irq_desc;
+	struct no_os_irq_init_param gpio_irq_desc_param = {
 		.irq_ctrl_id = GPIO_IRQ_ID,
 		.platform_ops = GPIO_IRQ_OPS,
 		.extra = GPIO_IRQ_EXTRA
 	};
 
-	ret = no_os_irq_ctrl_init(&global_desc, &global_desc_param);
+	/** MAX14906 Initialization */
+	ret = max14906_init(&max14906_desc, &max14906_ip);
 	if (ret)
-		goto max14906_remove;
+		goto exit;
 
-	ret = no_os_irq_set_priority(global_desc, GPIO_IRQ_ID, 1);
+	ret = no_os_irq_ctrl_init(&gpio_irq_desc, &gpio_irq_desc_param);
 	if (ret)
-		goto remove_gpio_irq_ctrl;
-
-	struct no_os_callback_desc gpio_cb = {
-		.callback = gpio_callback_fn,
-		/** Parameter to be passed when the callback is called. */
-		.ctx = max14906_desc,
-		.event = NO_OS_EVT_GPIO,
-		.peripheral = NO_OS_GPIO_IRQ,
-		.handle = NULL
-	};
+		goto max14906_remove;
 
-	ret = no_os_irq_register_callback(global_desc, GPIO_IRQ_ID,
+	ret = no_os_irq_register_callback(gpio_irq_desc, GPIO_FAULT_PIN_NUM,
 					  &gpio_cb);
 	if (ret)
 		goto remove_gpio_irq_ctrl;
 
-	ret = no_os_irq_trigger_level_set(global_desc, GPIO_IRQ_ID,
+	ret = no_os_irq_trigger_level_set(gpio_irq_desc, GPIO_FAULT_PIN_NUM,
 					  NO_OS_IRQ_EDGE_FALLING);
 	if (ret)
 		goto gpio_irq_unregister_callback;
 
-	ret = no_os_irq_enable(global_desc, GPIO_IRQ_ID);
+	ret = no_os_irq_set_priority(gpio_irq_desc, GPIO_FAULT_PIN_NUM, 1);
```

Comment:
I know this was also done before this commit, but why do you need to set the priority? Also what's the point of using the `gpio_irq_desc` instead?

Reply:
Updating the commit message.
And related to gpio_irq_desc, I thought it makes more sense for that descriptor to be named this way instead of global_desc.